operations/mimir-mixin/dashboards/reads.libsonnet

```
@@ -212,40 +212,80 @@ local filename = 'mimir-reads.json';
+            |||
+              kube_horizontalpodautoscaler_status_current_replicas{%(namespace_matcher)s, horizontalpodautoscaler=~"%(hpa_name)s"}
+              # HPA doesn't go to 0 replicas, so we multiply by 0 if the HPA is not active.
+              * on (%(cluster_labels)s, horizontalpodautoscaler)
+                kube_horizontalpodautoscaler_status_condition{%(namespace_matcher)s, horizontalpodautoscaler=~"%(hpa_name)s", condition="ScalingActive", status="true"}
+              # Add the scaletargetref_name label which is more readable than "kube-hpa-..."
+              + on (%(cluster_labels)s, horizontalpodautoscaler) group_left (scaletargetref_name)
+                0*kube_horizontalpodautoscaler_info{%(namespace_matcher)s, horizontalpodautoscaler=~"%(hpa_name)s"}
+            ||| % {
+              namespace_matcher: $.namespaceMatcher(),
+              cluster_labels: std.join(', ', $._config.cluster_labels),
+              hpa_name: $._config.autoscaling.querier.hpa_name,
+            },
           ],
           [
-            'Min',
-            'Max',
-            'Current',
+            'Max {{ scaletargetref_name }}',
+            'Current {{ scaletargetref_name }}',
           ],
         ) +
         $.panelDescription(
           title,
           |||
-            The minimum, maximum, and current number of querier replicas.
+            The maximum, and current number of querier replicas.
+            Please note that the current number of replicas can still show 1 replica even when scaled to 0.
+            Since HPA never reports 0 replicas, the query will report zero only if the HPA is not active.
           |||
-        ),
+        ) +
+        {
+          seriesOverrides+: [
+            {
+              alias: '/Max .+/',
+              dashes: true,
+              fill: 0,
+            },
+            {
+              alias: '/Current .+/',
+              fill: 0,
+            },
+          ],
+        }
       )
       .addPanel(
-        local title = 'Scaling metric';
+        local title = 'Scaling metric (desired replicas)';
         $.panel(title) +
         $.queryPanel(
           [
-            $.filterKedaMetricByHPA('keda_metrics_adapter_scaler_metrics_value', $._config.autoscaling.querier.hpa_name),
-            'kube_horizontalpodautoscaler_spec_target_metric{%s, horizontalpodautoscaler="%s"}' % [$.namespaceMatcher(), $._config.autoscaling.querier.hpa_name],
+            |||
+              keda_metrics_adapter_scaler_metrics_value
+              /
+              on(metric) group_left
+              label_replace(
+                  kube_horizontalpodautoscaler_spec_target_metric{%s, horizontalpodautoscaler=~"%s"},
+                  "metric", "$1", "metric_name", "(.+)"
+              )
+            ||| % [$.namespaceMatcher(), $._config.autoscaling.querier.hpa_name],
           ], [
-            'Scaling metric',
-            'Target per replica',
+            'Desired for {{ scaledObject }}',
```

nit: the panel name already helps for this
```suggestion
            '{{ scaledObject }}',
```

maybe the same comment for the "Current <>" replicas

---

Also, have you considered stacking the lines for the number of replicas? It'd be easier to see how many replicas there are in total at a glance

---

I tried stacking at the beginning, but then Grafana lies on the numbers.

If you stack `querier=2, querier-burst=0`, then when you hover on the lines, Grafana says `querier=2, querier-burst=2`.

---

Done the label change in  https://github.com/grafana/mimir/pull/3962/commits/734ccb49ddaf7be0723bbaf9e6081e5ecbc4c999